wolftpm/tpm2_wrap.h

```diff
@@ -118,214 +118,1699 @@ typedef struct WOLFTPM2_CAPS {
+
+    \param dev pointer to a TPM2_DEV struct
+    \param handle pointer to a structure of WOLFTPM2_HANDLE type, with a valid TPM 2.0 handle value
+
+    \sa wolfTPM2_Clear
+*/
 WOLFTPM_API int wolfTPM2_UnloadHandle(WOLFTPM2_DEV* dev, WOLFTPM2_HANDLE* handle);
 
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Deinitializes wolfTPM and wolfcrypt(if enabled)
+
+    \return TPM_RC_SUCCESS: successful
+    \return TPM_RC_FAILURE: generic failure (check TPM IO and TPM return code)
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+
+    \sa wolfTPM2_Clear
+*/
 WOLFTPM_API int wolfTPM2_Clear(WOLFTPM2_DEV* dev);
 
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Helper function to start a TPM generated hash
+
+    \return TPM_RC_SUCCESS: successful
+    \return TPM_RC_FAILURE: generic failure (check TPM IO and TPM return code)
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param hash pointer to a WOLFTPM2_HASH structure
+    \param hashAlg integer value, specifying a valid TPM 2.0 hash algorithm
+    \param usageAuth pointer to a string constant, specifying the authorization for subsequent use of the hash
+    \param usageAuthSz integer value, specifying the size of the authorization, in bytes
+
+    \sa wolfTPM2_HashUpdate
+    \sa wolfTPM2_HashFinish
+*/
 WOLFTPM_API int wolfTPM2_HashStart(WOLFTPM2_DEV* dev, WOLFTPM2_HASH* hash,
     TPMI_ALG_HASH hashAlg, const byte* usageAuth, word32 usageAuthSz);
+
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Update a TPM generated hash with new user data
+    Note: Make sure the auth is correctly set
+
+    \return TPM_RC_SUCCESS: successful
+    \return TPM_RC_FAILURE: generic failure (check TPM IO and TPM return code)
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param hash pointer to a WOLFTPM2_HASH structure
+    \param data pointer to a byte buffer, containing the user data to be added to the hash
+    \param dataSz integer value of word32 type, specifying the size of the user data, in bytes
+
+    \sa wolfTPM2_HashStart
+    \sa wolfTPM2_HashFinish
+*/
 WOLFTPM_API int wolfTPM2_HashUpdate(WOLFTPM2_DEV* dev, WOLFTPM2_HASH* hash,
     const byte* data, word32 dataSz);
+
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Finalize a TPM generated hash and get the digest output in a user buffer
+    Note: Make sure the auth is correctly set
+
+    \return TPM_RC_SUCCESS: successful
+    \return TPM_RC_FAILURE: generic failure (check TPM IO and TPM return code)
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param hash pointer to a WOLFTPM2_HASH structure
+    \param digest pointer to a byte buffer, used to store the resulting digest
+    \param digestSz integer value of word32 type, specifying the size of the digest, in bytes
```

Comment:
Also effects other hashing `*Finish()` calls
```suggestion
    \param[inout] digestSz pointer to size of digest buffer, on return set to bytes stored in digest buffer
```